docs/generate_examples.cmake

```diff
@@ -0,0 +1,48 @@
+
+# Get and update the LUA_PATH so the scripts can be executed without Awesome
+execute_process(COMMAND lua -e print\(package.path\) OUTPUT_VARIABLE LUA_PATH_)
+set(ENV{LUA_PATH} "${SOURCE_DIR}/lib/?;${SOURCE_DIR}/lib/?.lua;${LUA_PATH_}")
+
+execute_process(COMMAND mkdir -p "${BUILD_DIR}/doc/images")
+
+#
+# Shape API
+#
+foreach (SHAPE_NAME "circle" "arrow" "rounded_rect" "hexagon" "infobubble"
+    "powerline" "isosceles_triangle" "cross" "octogon" "parallelogram"
+    "losange" "partially_rounded_rect" "radial_progress" "rounded_bar"
+    "rectangle" "rectangular_tag")
+    set(SHAPE_FILE "${SOURCE_DIR}/tests/shape/${SHAPE_NAME}.lua")
+    set(SHAPE_SVG  "${BUILD_DIR}/doc/images/shape_${SHAPE_NAME}.svg")
+
+    # Generate some SVG for the documentation and load the examples for the doc
+    execute_process(
+        COMMAND lua ${SOURCE_DIR}/tests/shape/test-shape.lua ${SHAPE_FILE} ${SHAPE_SVG}
+        OUTPUT_VARIABLE SHAPE_OUTPUT
+        ERROR_VARIABLE  SHAPE_ERROR
+    )
+
+    if (NOT SHAPE_ERROR STREQUAL "")
+        message(${SHAPE_ERROR})
+        message(${SHAPE_NAME} " SVG generation failed, bye")
+        exit(1)
```

Comment:
Umm, good catch, . It was working, yes, but not the way it was intended to (Aka, it did call `message(FATAL_ERROR)` "indirectly".